src/service/balance-service.ts

```diff
@@ -380,4 +381,30 @@ export default class BalanceService extends WithManager {
   public async getBalance(id: number, date?: Date): Promise<BalanceResponse> {
     return (await this.getBalances({ ids: [id], allowDeleted: true, date })).records[0];
   }
+
+  /**
+   * Get the total balance of SudoSOS
+   * @param allUsers Whether or not you want to get the balance for all users or only the users that can top up.
+   * @param date Date to check the balances for
+   */
+  public async calculateTotalBalances(allUsers: boolean, date: Date): Promise<TotalBalanceResponse> {
+    const userIds = allUsers ? (await User.find()).map((user) => user.id)
```

Comment:
I think a single `await User.find()` would be nicer, and instead making the where a conditional, as this is quite hard  to read.

```ts
const where = allUsers ? {} : { type: In([UserType.LOCAL_USER, UserType.MEMBER]) }
```